src/telemetry/mod.rs

```diff
@@ -19,23 +51,34 @@ pub mod metrics {
             Unit::Count,
             "Total number of requests received by the web server"
         );
+
+        describe_counter!(
+            RequestProtocolVersion::from(Version::HTTP_11),
+            Unit::Count,
+            "Total number of HTTP/1.1 requests received"
+        );
+
+        describe_counter!(
+            RequestProtocolVersion::from(Version::HTTP_2),
+            Unit::Count,
+            "Total number of HTTP/2 requests received"
+        );
     }
 
     #[cfg(test)]
     #[must_use]
-    pub fn get_counter_value(
+    pub fn get_counter_value<K: Into<KeyName>>(
         snapshot: metrics_util::debugging::Snapshot,
-        metric_name: &'static str,
+        metric_name: K,
     ) -> Option<u64> {
         use metrics_util::debugging::DebugValue;
         use metrics_util::MetricKind;
 
         let snapshot = snapshot.into_vec();
+        let metric_name = metric_name.into();
```

Comment:
confusing that this works... is it looking at `as_str()` and somehow realizing that `KeyName` is the exclusive conversion that has this function? what if there were 2 possible `Into<_>` that implemented `as_str()`?

Reply:
I think compiler uses the knowledge that `K: Into<KeyName>` to resolve the ambiguity here

Reply:
huh so it conditionally compiles based on how many implicit conversions are in scope